### PR TITLE
[Merged by Bors] - feat(Order/Floor): `x - 1 / 2 < round x`

### PR DESCRIPTION
**Pull request by Colin166**
Adding a new theorem to Mathlib/Algebra/Order/Floor.lean : 

```
theorem round_half_bot : a - (1 / 2 : ℚ) < round a := by
  rw [round_eq, show (a - 1 / 2 : ℚ) = (a + 1 / 2) - 1 by ring]
  exact Int.sub_one_lt_floor (a + 1 / 2)
```

### CODE REVIEW

File: Mathlib/Algebra/Order/Floor.lean [hunk 1]
```diff
@@ -1340,6 +1341,16 @@ theorem round_le (x : α) (z : ℤ) : |x - round x| ≤ |x - z| := by
     norm_cast
     exact floor_le_sub_one_iff.mpr hx
 
+@[simp]
+theorem round_half_bot [HSub α ℝ α] : x - (1 / 2 : ℝ) < round x := by
+  rw [round_eq x, show (x - 1 / 2 : ℝ) = (x + 1 / 2) - 1 by ring]
```
Comment (Command-Master) [inline Mathlib/Algebra/Order/Floor.lean:1346]:
```suggestion
@[simp]
theorem round_half_bot : x - (1 / 2 : ℚ) < round x := by
  rw [round_eq x, show x - (1 / 2 : ℚ) = x + 1 / 2 - 1 by ring]
```
You shouldn't use `HSub α ℝ α`, the subtraction it defines is unconnected to `α`'s field structure, you should use the coercion from `ℚ` to `α` (which is part of `α`'s field structure).
Additionally I'm not sure it should be a `simp` lemma. Is there any reason for that?

File: Mathlib/Algebra/Order/Floor.lean [hunk 2]
```diff
@@ -1340,6 +1341,16 @@ theorem round_le (x : α) (z : ℤ) : |x - round x| ≤ |x - z| := by
...
+  exact Int.sub_one_lt_floor (x + 1 / 2)
+
+@[simp]
+theorem round_half_top (x : α) [HAdd α ℝ α] : (round x : α) ≤ x + (1 / 2 : ℝ) := by
```
Comment (Command-Master) [inline Mathlib/Algebra/Order/Floor.lean:1350]:
```suggestion
@[simp]
theorem round_half_top (x : α) : (round x : α) ≤ x + (1 / 2 : ℚ) := by
```

same here.

File: Mathlib/Algebra/Order/Floor.lean [hunk 3]
```diff
@@ -8,6 +8,7 @@ import Mathlib.Algebra.Order.Interval.Set.Group
 import Mathlib.Algebra.Group.Int
 import Mathlib.Data.Int.Lemmas
 import Mathlib.Data.Nat.Cast.Order.Field
+import Mathlib.Data.Real.Basic
```
Comment (Command-Master) [inline Mathlib/Algebra/Order/Floor.lean:11]:
```suggestion
```
This import should be removed.

**Review by Command-Master — CHANGES_REQUESTED** [review]

**Comment by Command-Master** [comment]
Additionally both theorems should be moved to the `LinearOrderedField` section, after `round_eq`

File: Mathlib/Algebra/Order/Floor.lean (hunk 3 above)
Comment (Colin166) [reply Mathlib/Algebra/Order/Floor.lean]: Fixed everything

File: Mathlib/Algebra/Order/Floor.lean [hunk 4]
```diff
@@ -1395,6 +1396,14 @@ theorem abs_sub_round_div_natCast_eq {m n : ℕ} :
   rw [abs_sub_round_eq_min, Nat.cast_min, ← min_div_div_right hn'.le,
     fract_div_natCast_eq_div_natCast_mod, Nat.cast_sub (m.mod_lt hn).le, sub_div, div_self hn'.ne']
 
+theorem round_half_bot (x : α) : x - 1 / 2 < round x := by
+  rw [round_eq x, show x - 1 / 2 = x + 1 / 2 - 1 by ring]
```
Comment (Command-Master) [inline Mathlib/Algebra/Order/Floor.lean:1400]:
```suggestion
  rw [round_eq x, show x - 1 / 2 = x + 1 / 2 - 1 by linarith]
```
letting linarith prove this avoids having to import `Tactic.Ring`

File: Mathlib/Algebra/Order/Floor.lean [hunk 5]
```diff
@@ -14,6 +14,7 @@ import Mathlib.Order.GaloisConnection
 import Mathlib.Tactic.Abel
 import Mathlib.Tactic.Linarith
 import Mathlib.Tactic.Positivity
+import Mathlib.Tactic.Ring
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Floor.lean:17]: Shouldn't be necessary to add that
```suggestion
```

File: Mathlib/Algebra/Order/Floor.lean [hunk 6]
```diff
@@ -1395,6 +1396,14 @@ theorem abs_sub_round_div_natCast_eq {m n : ℕ} :
   rw [abs_sub_round_eq_min, Nat.cast_min, ← min_div_div_right hn'.le,
     fract_div_natCast_eq_div_natCast_mod, Nat.cast_sub (m.mod_lt hn).le, sub_div, div_self hn'.ne']
 
+theorem round_half_bot (x : α) : x - 1 / 2 < round x := by
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Floor.lean:1399]: Please use descriptive names follow the [naming convention](https://leanprover-community.github.io/contribute/naming.html)
```suggestion
theorem sub_half_lt_round (x : α) : x - 1 / 2 < round x := by
```
`bot` refers to `⊥` and there is no `⊥` in your theorem statement

File: Mathlib/Algebra/Order/Floor.lean [hunk 7]
```diff
@@ -1395,6 +1396,14 @@ theorem abs_sub_round_div_natCast_eq {m n : ℕ} :
...
+  rw [round_eq x, show x - 1 / 2 = x + 1 / 2 - 1 by ring]
+  exact Int.sub_one_lt_floor (x + 1 / 2)
+
+theorem round_half_top (x : α) : (round x : α) ≤ x + 1 / 2 := by
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Floor.lean:1403]: I will let you find the correct name of this one as an exercise

File: Mathlib/Algebra/Order/Floor.lean (hunk 4 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Floor.lean:1400]: FYI `linarith` uses `ring` under the hood, so your comment is actually to replace a tactic by a more expensive tactic.

Comment (Command-Master) [reply Mathlib/Algebra/Order/Floor.lean:1400]: Are you sure? Importing `Mathlib.Tactic.Linarith` doesn't let me use the `ring` tactic

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Floor.lean:1400]: See for yourself: https://github.com/leanprover-community/mathlib4/blob/8973ec00e9e76fc00ed67953e0ca960781f76e4c/Mathlib/Tactic/Linarith/Datatypes.lean#L7

**Comment by semorrison** [comment]
This pulls in additional imports to a file making basic definitions. Please move to a later or new file.

File: Mathlib/Algebra/Order/Floor.lean (hunk 7 above)
Comment (Colin166) [reply Mathlib/Algebra/Order/Floor.lean]: I renamed it to `round_le_add_half`. Is that correct?

**Comment by Colin166** [comment]
Without `import Mathlib.Tactic.Ring` Lean doesn't recognize `ring` as a valid tactic; that's why I included it. I will try to prove the goal in a different way.

**Comment by Colin166** [comment]
Alright issue should be fixed. I replaced `ring` with `nlinarith`

File: Mathlib/Algebra/Order/Floor.lean (hunk 7 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Floor.lean]: Yes!

File: Mathlib/Algebra/Order/Floor.lean [hunk 8]
```diff
@@ -1395,6 +1395,14 @@ theorem abs_sub_round_div_natCast_eq {m n : ℕ} :
   rw [abs_sub_round_eq_min, Nat.cast_min, ← min_div_div_right hn'.le,
     fract_div_natCast_eq_div_natCast_mod, Nat.cast_sub (m.mod_lt hn).le, sub_div, div_self hn'.ne']
 
+theorem sub_half_lt_round (x : α) : x - 1 / 2 < round x := by
+  rw [round_eq x, show x - 1 / 2 = x + 1 / 2 - 1 by nlinarith]
+  exact Int.sub_one_lt_floor (x + 1 / 2)
+
+theorem round_le_add_half (x : α) : (round x : α) ≤ x + 1 / 2 := by
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Floor.lean:1402]: This type-ascription shoiuld be unnecessary
```suggestion
theorem round_le_add_half (x : α) : round x ≤ x + 1 / 2 := by
```

**Review by YaelDillies — COMMENTED** [review]
Thanks!

maintainer merge

**Comment by urkud** [comment]
Please update the commit message to reflect name changes. Otherwise LGTM. Thanks!
bors d+

**Pull request closed without merging**